### PR TITLE
WS-3069: add taxi_form_id and taxi_form_grouping fields to Degree model

### PR DESCRIPTION
**Pull request by marekwrobel**
[WS-3069](https://2u-internal.atlassian.net/browse/WS-3069) requests that editors would be able to specify a custom url for a form to be rendered in an iframe in place of the current hubspot form on masters pages. An example page with that form in the sidebar: https://www.edx.org/masters/online-master-business-administration-mba-bux.

In order for this feature to be provided, a new field had to be created for the Degree model.

[A full discussion](https://edx-internal.slack.com/archives/C032DFXHV7W/p1657645219748239) took place over this subject where a few things were laid out:

- Taxi Forms are an in-house form engine that's built by 2U.
- These new fields are not just for 2U degrees, but rather all Degrees.

### CODE REVIEW

**Comment by danielkalen** [comment]
This is to support this prospectus ticket: https://github.com/edx/prospectus/pull/4220

File: course_discovery/apps/course_metadata/models.py [hunk 1]
```diff
@@ -2859,7 +2859,16 @@ class Degree(Program):
...
+    taxi_form_id = models.URLField(
+        help_text=_('The ID of the 2u Taxi form that would be rendered in place of the hubspot capture form.'),
+        null=True,
+        blank=True,
+    )
+    taxi_form_grouping = models.CharField(
+        help_text=_('The grouping of the 2u taxi form that would be rendered in place of the hubspot capture form.'),
+        max_length=50,
+        blank=True,
+    )
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/models.py]: If these fields are meant for 2U Degrees only, please add them to DegreeAdditionalMetadata model.

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +1,4 @@
 # This file is here because many Platforms as a Service look for
 #	requirements.txt in the root directory of a project.
 -r requirements/production.txt
+
```
Comment (DawoudSheraz) [inline requirements.txt]: nit: to remove this new line change.

Comment (danielkalen) [reply requirements.txt]: had do do that in order to trigger a build

File: course_discovery/apps/course_metadata/models.py (hunk 1 above)
Comment (danielkalen) [reply course_discovery/apps/course_metadata/models.py]: These are not for 2U degress only, this discussion already took place: https://edx-internal.slack.com/archives/C032DFXHV7W/p1657807820241819?thread_ts=1657645219.748239&cid=C032DFXHV7W

File: course_discovery/apps/course_metadata/models.py [hunk 3]
```diff
@@ -2859,7 +2859,16 @@ class Degree(Program):
         blank=True,
         max_length=128,
     )
-
+    taxi_form_id = models.URLField(
+        help_text=_('The ID of the 2u Taxi form that would be rendered in place of the hubspot capture form.'),
+        null=True,
+        blank=True,
+    )
```
Comment (azanbinzahid) [inline course_discovery/apps/course_metadata/models.py]: Is it the ID or the URL? please correct field type or help_text accordingly.

**Review by azanbinzahid — COMMENTED** [review]
- Please update PR description, add context from [this](https://edx-internal.slack.com/archives/C032DFXHV7W/p1657645219748239) thread.

File: course_discovery/apps/course_metadata/models.py (hunk 3 above)
Comment (DawoudSheraz) [reply course_discovery/apps/course_metadata/models.py]: +1 on this.

**Review by DawoudSheraz — COMMENTED** [review]
Please update PR description to add more details.

File: course_discovery/apps/course_metadata/models.py (hunk 3 above)
Comment (danielkalen) [reply course_discovery/apps/course_metadata/models.py]: Nice catch!

**Comment by danielkalen** [comment]
[WS-3069](https://2u-internal.atlassian.net/browse/WS-3069) requests that editors would be able to specify a custom url for a form to be rendered in an iframe in place of the current hubspot form on masters pages. An example page with that form in the sidebar: https://www.edx.org/masters/online-master-business-administration-mba-bux.

In order for this feature to be provided, a new field had to be created for the Degree model.

[A full discussion](https://edx-internal.slack.com/archives/C032DFXHV7W/p1657645219748239) took place over this subject where a few things were laid out:

- Taxi Forms are an in-house form engine that's built by 2U.
- These new fields are not just for 2U degrees, but rather all Degrees.

**Review by DawoudSheraz — APPROVED** [review]

**Pull request merged**